src/App.jsx

```diff
@@ -1,50 +1,45 @@
-import logo from "/logo.png";
 import "./App.css";
-import { onChildAdded, push, ref, set } from "firebase/database";
-import { database } from "./firebase";
-import { useState, useEffect } from "react";
-
-// Save the Firebase message folder name as a constant to avoid bugs due to misspelling
-const DB_MESSAGES_KEY = "messages";
+import { useState } from "react";
+import LoginSignUp from "./Components/LoginSignup";
+import PostDisplay from "./Components/PostDisplay";
+import { createBrowserRouter, RouterProvider } from "react-router-dom";
+import Navbar from "./Components/Navbar";
 
 function App() {
-  const [messages, setMessages] = useState([]);
-
-  useEffect(() => {
-    const messagesRef = ref(database, DB_MESSAGES_KEY);
-    // onChildAdded will return data for every child at the reference and every subsequent new child
-    onChildAdded(messagesRef, (data) => {
-      // Add the subsequent child to local component state, initialising a new array to trigger re-render
-      setMessages((prevState) =>
-        // Store message key so we can use it as a key in our list items when rendering messages
-        [...prevState, { key: data.key, val: data.val() }]
-      );
-    });
-  }, []);
+  const [isLoggedIn, setIsLoggedIn] = useState(false);
+  const [user, setUser] = useState({});
 
-  const writeData = () => {
-    const messageListRef = ref(database, DB_MESSAGES_KEY);
-    const newMessageRef = push(messageListRef);
-    set(newMessageRef, "abc");
-  };
-
-  // Convert messages in state to message JSX elements to render
-  let messageListItems = messages.map((message) => (
-    <li key={message.key}>{message.val}</li>
-  ));
+  const router = createBrowserRouter([
+    {
+      path: "/",
+      element: (
+        <div>
+          <Navbar />
+          <br />
+          <LoginSignUp setUser={setUser} setIsLoggedIn={setIsLoggedIn} />
+        </div>
+      ),
+    },
+    {
+      path: "/PostsDisplay",
+      element: (
+        <div>
+          <Navbar />
+          <br />
+          <PostDisplay setUser={setUser} user={user} setIsLoggedIn={setIsLoggedIn} isLoggedIn={isLoggedIn} />
+        </div>
+      ),
+    },
+  ]);
 
   return (
-    <>
-      <div>
-        <img src={logo} className="logo" alt="Rocket logo" />
-      </div>
-      <h1>Instagram Bootcamp</h1>
-      <div className="card">
-        {/* TODO: Add input field and add text input as messages in Firebase */}
-        <button onClick={writeData}>Send</button>
-        <ol>{messageListItems}</ol>
-      </div>
-    </>
+    <div className="bg-gradient-to-b from-cyan-300 to-white-100 shadow-2xl dark:bg-grey-400 px-3 py-2 rounded-lg">
+      <h1 className="text-2xl font-bold text-rose-500 mb-4">Rocketgram</h1>
+
```

Comment:
Generally and conventionally the RouterProvider is a self closing element:
<RouterProvider
    router={router}
  />